zulip_bots/zulip_bots/bots/idonethis/idonethis.py

```diff
@@ -0,0 +1,211 @@
+    else:
+        return make_API_request("/entries".format(team_id))
+
+def api_create_entry(body: str, team_id: str) -> Any:
+    return make_API_request("/entries", "POST", {"body": body, "team_id": team_id})
+
+def list_steams() -> str:
+    data = api_list_team()
+    response = "Teams:\n"
+    for team in data:
+        response += " * " + team['name'] + "\n"
+    return response
+
+def get_team_hash(team_name: str) -> str:
+    data = api_list_team()
+    for team in data:
+        if team['name'].lower() == team_name.lower() or team['hash_id'] == team_name:
+            return team['hash_id']
+    raise TeamNotFoundException(team_name)
+
+def team_info(team_name: str) -> str:
+    data = api_show_team(get_team_hash(team_name))
+    response = "Team Name: " + data['name'] + "\n"
+    response += "ID: `" + data['hash_id'] + "`\n"
+    response += "Created at: " + data['created_at'] + "\n"
+    return response
+
+def entries_list(team_name: str) -> str:
+    if team_name:
+        data = api_list_entries(get_team_hash(team_name))
+    else:
+        data = api_list_entries()
+    response = "Entries for " + team_name + ":\n"
+    for entry in data:
+        response += " * " + entry['body_formatted'] + "\n"
+        response += "  * Created at: " + entry['created_at'] + "\n"
+        response += "  * Status: " + entry['status'] + "\n"
+        response += "  * User: " + entry['user']['full_name'] + "\n"
+        response += "  * Team: " + entry['team']['name'] + "\n"
+        response += "  * ID: " + entry['hash_id'] + "\n"
+    return response
+
+def create_entry(message: str) -> str:
+    SINGLE_WORD_REGEX = re.compile("--team=([a-zA-Z0-9_]*)")
+    MULTIWORD_REGEX = re.compile('"--team=([^"]*)"')
+
+    team = ""
+    new_message = ""
+    single_word_match = SINGLE_WORD_REGEX.search(message)
+    multiword_match = MULTIWORD_REGEX.search(message)
+
+    if multiword_match is not None:
+        team = multiword_match.group(1)
+        new_message = MULTIWORD_REGEX.sub("", message).strip()
+    elif single_word_match is not None:
+        team = single_word_match.group(1)
+        new_message = SINGLE_WORD_REGEX.sub("", message).strip()
+    elif default_team:
+        team = default_team
+        new_message = message
+    else:
+        raise UnknownCommandSyntax("""I don't know which team you meant for me to create an entry under.
+Either set a default team or pass the `--team` flag.
+More information in my help""")
+
+    team_id = get_team_hash(team)
+    data = api_create_entry(new_message, team_id)
+    return "Great work :thumbs_up:. New entry `{}` created!".format(data['body_formatted'])
+
+class IDoneThisHandler(object):
+    def initialize(self, bot_handler: Any) -> None:
+        global api_key, default_team
+        self.config_info = bot_handler.get_config_info('idonethis')
+        if 'key' in self.config_info:
+            api_key = self.config_info['key']
```

Comment:
This key is required, right? You should quit the bot with an error if there is no api key.